src/RunWasmClient.ts

```diff
@@ -66,15 +68,38 @@ export class TSClient {
         this.logs.push(value)
       }
     }
+
+    const libs: string[] = ['es5', 'dom']
+
+    const fetchLibs = async (libs: string[]) => {
+      return Promise.all(
+        libs.map(async (lib: string) => {
+          const path = `typescript/lib/lib.${lib}.d.ts`
+          const content = await fetch(
+            `https://cdn.jsdelivr.net/npm/typescript@4.4.3/lib/lib.${lib}.d.ts`
+          )
+            .then((res) => res.text())
+            .catch(() => console.log('Failed to load lib: ' + lib))
+
+          return {
+            path,
+            content,
+            ast: ts.createSourceFile(path, content, ts.ScriptTarget.Latest),
+          }
+        })
+      ).then((libs) => libs)
+    }
+
+    this.libData = fetchLibs(libs)
   }
 
   public async run({
     code,
   }: {
     code: string
   }): Promise<{ errors: string[]; output: string[] }> {
-    const typeErrors = getTSTypeErrors(code, this.ts)
-
+    const libData = await this.libData
```

Comment:
This is my concern is that. Every time when method `run` will be called we are loading the lib which can be slow.
I don't know the solution but is it possible that we approach it in such a way that the lib loads only one time?
As with the current approach when every user will click `Run Code` this lib will load.

This is my observation, I can be wrong though so forgive me for it.

Reply:
Yes, I observed this as well. That's a byproduct of the fact that we create a new client each time we run the code. In order to change this the way that the end-user interacts with the run-wasm library would have to change a bit. I considered this out of the scope of the issue I was working on so I didn't work on it, but if that's something the maintainers would like to pursue here I can look at that.

Reply:
oh yeah, I see it now.
I missed it this line how the library is used 
`let tsClient = createTSClient(window.ts)`

Reply:
Thanks for taking this on! Functionality wise this looks great, `Date` and `JSON` are working for me :) IMO the latency of making these lib calls on every call to `run` is a bit too high though, given that we know they won't change. I think it'd be worth making the change to only call the `createTSClient` once in the page, which FWIW is what we do with the Python page. As you say this is a change, but it'll actually align the TS page better with the Python one and is IMO a better approach as soon as we introduce this sort of pre-loading. 

I think this would also simplify the code - the constructor can stay simple, your `fetchLibs` can become a public function, and the Next page can run it before enabling the run button so it just becomes part of the initial load.

This would mean that we only ever call `fetchLibs` once, and that when the run button is enabled everything is already loaded and will be very snappy.

If it'd be helpful I'd be happy to open a PR with the change to the TS page to only create the client once. Since you haven't changed that file at all we won't conflict.

Reply:
@mcintyre94 So you would adjust the next-js page to only create the client once, and I would adjust the fetch code to clean up with that adjustment made?

Reply:
If that works for you! I think it’d be worth it for the performance improvement :)

Reply:
Just opened a PR: #80 Let me know how it looks/if it makes sense to you! :)

Reply:
Yes, it makes perfect sense. I'm not sure about the best way to refactor the TSClient to clean it up though. As it is it already will only load once (from the constructor), and on future calls to run it will only actually have to wait for an asynchronous call if its the first time code is being run. You mentioned making fetchLibs public (and thus moving it out of the constructor), does this mean that the intention is to make a call to it from the Next page when you load it up?

Reply:
That's a great point actually, nice! That was my thinking yep, there's a commented out line in my PR showing it: `tsClient.fetchLibs(['es5', 'dom']).then(() => setTsClient(tsClient))`

It definitely doesn't make as much of a difference if it's only on first run though like you said. IMO it'd be preferable to have it on load, just to avoid slowing the user's first run. But not sure if that's a blocker really.

Reply:
@mcintyre94 I agree with this approach. It won't be a noticeable delay anyway & will make all runs faster

Reply:
@mcintyre94 Sounds good I will update my branch to make fetchLibs part of the public interface!